### PR TITLE
[Poetry] Implement profanity filtering in custom poem editor

### PR DESCRIPTION
**Pull request by maddiedierker**
Implements profanity filtering for the custom poem editor, plus some small related UX tweaks to the dismiss behavior of the dialog.

If the title, author, or poem fields contain profanity, we'll prevent the poem from saving and show an error message so the user can fix their poem:
Screen Shot 2021-10-14 at 5 37 15 PM

UX tweaks:
- If there's an error (errors only happen when there's profanity right now) when the dialog closes, the dialog state (title, author, poem) will be cleared so the dialog is empty upon re-opening. Otherwise, we'll keep the state so the user can choose to continue editing the poem upon re-open.
- Only update the selected poem in the project/redux if we receive a poem when the dialog closes. If the user dismisses the dialog (e.g., clicks the "X" or clicks away instead of "Save"), we'll no-op and keep the originally-selected poem.

## Links

- [STAR-1763](https://codedotorg.atlassian.net/browse/STAR-1763)

## Testing story

Adds unit tests.

### CODE REVIEW

File: apps/src/componentLibrary/StylizedBaseDialog.jsx [hunk 1]
```diff
@@ -169,7 +169,7 @@ const styles = {
     padding: `${GUTTER / 2}px ${GUTTER - 3}px` // -3 to account for 3px-margin around <Button/>
   },
   buttons: {
-    all: {boxShadow: 'none'},
+    all: {boxShadow: 'none', flexShrink: 0},
```
Comment (maddiedierker) [inline apps/src/componentLibrary/StylizedBaseDialog.jsx:172]: Fixes a UI bug where really long footer content would shrink the FooterButton(s):
Screen Shot 2021-10-13 at 5 18 28 PM

File: apps/src/p5lab/poetry/PoemSelector.jsx [hunk 2]
```diff
@@ -1,42 +1,61 @@
...
+    }
+
+    setError(null);
+  }, [props.isOpen]);
 
   const body = (
     <div>
       <div style={styles.modalContainer}>
-        <label style={styles.label}>{`${msg.title()}: `}</label>
+        <label style={styles.label}>{msg.title()}</label>
```
Comment (maddiedierker) [inline apps/src/p5lab/poetry/PoemSelector.jsx:42]: moves colons on labels to i18n for RTL languages

File: apps/src/p5lab/poetry/constants.js [hunk 3]
```diff
@@ -1,3 +1,8 @@
+export const PoetryStandaloneApp = {
+  Poetry: 'poetry',
+  PoetryHoc: 'poetry_hoc'
```
Comment (maddiedierker) [inline apps/src/p5lab/poetry/constants.js:3]: looking for feedback: is this choice to change the casing confusing? i could make the enum match exactly:

```javascript
export const PoetryStandaloneApp = makeEnum('poetry', 'poetry_hoc');

// usage
PoetryStandaloneApp.poetry_hoc
```

but mixing ruby and js conventions makes me sad 😆

File: apps/i18n/poetry/en_us.json [hunk 4]
```diff
@@ -37,7 +37,8 @@
   "oneTextEffect": "Great work! You added one text effect. Try adding another.",
   "poLines": "The birds have vanished into the sky,\nAnd now the last cloud drains away.\nWe sit together, the mountain and me,\nUntil only the mountain remains.",
   "poTitle": "Mountain",
-  "poem": "Poem",
+  "poem": "Poem:",
+  "profanityFoundError": "Your poem text contains inappropriate language so it will not be saved. Please update your poem to remove the {wordCount, plural, one {word} other {words}} \"{words}\".",
```
Comment (epeach) [inline apps/i18n/poetry/en_us.json]: The word "Text" kind of sticks out to me. If I got this I would look at the poem content, even though the word might be in the title. Could we just remove the word 'text'?

File: apps/src/p5lab/poetry/constants.js (hunk 3 above)
Comment (epeach) [reply apps/src/p5lab/poetry/constants.js:3]: I like this change. I think it reads better. :+1:

**Review by epeach — APPROVED** [review]
I have one small nit comment about the text itself, but other than that, it looks good!

**Pull request merged**